crypto/bio/bss_dgram.c

```diff
@@ -225,8 +225,10 @@ static void dgram_adjust_rcv_timeout(BIO *b)
         if (getsockopt(b->num, SOL_SOCKET, SO_RCVTIMEO,
                        &(data->socket_timeout), (void *)&sz) < 0) {
             perror("getsockopt");
-        } else if (sizeof(sz.s) != sizeof(sz.i) && sz.i == 0)
-            OPENSSL_assert(sz.s <= sizeof(data->socket_timeout));
+        } else if (sizeof(sz.s) != sizeof(sz.i) && sz.i == 0) {
+            if (!ossl_assert(sz.s <= sizeof(data->socket_timeout)))
+                return 0;
+        }
```

Comment:
The purpose of this exercise should not to replace literally every OPENSSL_asserts, but those that are unjustified. Most notably if there is reason to believe that memory is corrupted, crashing application would be more appropriate than to return error. This check is actually guarding memory corruption. It verifies that optlen argument to getsockopt is not grossly misinterpreted by big-endian kernel in case when socklen_t is not int.